src/main/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImpl.java

```diff
@@ -81,12 +81,21 @@ public void saveRoom(Room room) {
     }
 
     @Override
-    public List<Room> getAllRooms(Pageable pageable) {
+    public List<Room> getAllRooms(String type, Pageable pageable) {
         Page<RoomEntity> roomEntityList = roomRepository.findAll(pageable);
```

Comment:
You can put  'if' here and in roomRepository create two method e.g.: one - findAllPlantRoom, second - findAllPetRoom, the sql query should grab all rooms.